### ISSUE TITLE
size mismatch on backwards for torch

### ISSUE THREAD

**Issue by ttdd11** [isarandi/nlf#27]
Amazing work.

I wrote a pytorch training script based on the tf version and it seems to work correctly on the forward but on the backwards, once   the loss shifts to reconstruct_ref_fullpersp, the backwards has a shape mismatch as a result of torch.linalg.lstsq. Something like: Check failed: dim1 == dim2 || dim1 == 1 || dim2 == 1 || dim1 

Are you seeing this as well? 

Thank you again for the great contribution and exceptional work.

**Comment by isarandi** [comment]
I have a new training code which works, but I need to do more code cleanups and tests. I'm seeing some instabilities and wanted to make sure everything is properly set up with gradscaler, multigpu, autocasting, checkpointing etc. But it's coming.

**Comment by ttdd11** [comment]
@isarandi alright that sounds great - does that new training code by chance have changes to reconstruct_ref_fullpersp?

If so, I'd happily test it over here to see if this resolves the issue I'm seeing!

Thanks again for the quick reply and great work on this.

**Comment by ttdd11** [comment]
@isarandi I did some more testing and it looks like its specific to the GPU and does not occur on the CPU! If you want some insight into this I'm happy to help.
